shopinvader_api_address/models/res_partner.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from fastapi import status
+from fastapi.exceptions import HTTPException
+
+from odoo import _, api, models
+
+from ..schema.schema import AddressInput, AddressSearch
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    @api.model
+    def _create_vals_shopinvader_address(self, data: AddressInput) -> dict:
+        state_id = self.env["res.country.state"].search(
+            [
+                "|",
+                ("name", "=", data.state),
+                ("code", "=", data.state),
+            ],
+            limit=1,
+        )
+
+        country_id = self.env["res.country"].search(
+            [
+                "|",
+                ("name", "=", data.country),
+                ("code", "=", data.country),
+            ]
+        )
```

Comment:
Sates and countries are translated fields. I would like to not rely on a search to get the id. IMO we should provide the available values (id, name) via a dedicated router.

Reply:
ok, so, we would need one "shopinvader_api_country",  "shopinvader_api_state" and here, I would directly use the id of state/country?